bionic/protocols.py

```diff
@@ -314,6 +317,45 @@ def write(self, array, path):
             np.save(file_, array)
 
 
+dd = import_optional_dependency('dask.dataframe', raise_on_missing=False)
+
+
+class DaskProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values always have the
+    ``dask.dataframe.DataFrame`` type.
+
+    These values will be serialized to .dask.pq files.
+    """
+
+    def get_fixed_file_extension(self):
+        return 'pq.dask'
+
+    def validate(self, value):
+        # If dd is None, then dask with dataframe (i.e. dask[dataframe]) is not present,
+        # which presumably means this value is not a dask dataframe, and hence should fail
+        # validation.
+        assert dd is not None
+        assert isinstance(value, dd.DataFrame)
+
+    def read(self, path, extension):
+        if six.PY2:
+            path = str(path)
+        with warnings.catch_warnings():
+            dd = import_optional_dependency('dask.dataframe', purpose='the @dask decorator')
```

Comment:
nit: this should probably be outside the `with` block, right?

Reply:
Yeah, prob best to have only the stuff that has to be in the `with`, changing!